### PR TITLE
EMS compatible with Qwen-MoE.

### PR DESCRIPTION
**Pull request by adoda**

### CODE REVIEW

File: chatlearn/models/megatron/memory_manager/trainer_v4.py [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Trainer Memery manager for Megatron V4"""
+from chatlearn.utils.megatron_import_memory_helper import MegatronVersion, check_megatron_versions
+
+check_megatron_versions([MegatronVersion.V4])
+
+# pylint: disable=wrong-import-position,wrong-import-order,ungrouped-imports
+from typing import List, Optional
```
Comment (SeaOfOcean) [inline chatlearn/models/megatron/memory_manager/trainer_v4.py:21]: 这个地方修一下import顺序？

Comment (adoda) [reply chatlearn/models/megatron/memory_manager/trainer_v4.py:17]: done

**Review by haolin-nju — APPROVED** [review]
LGTM

**Review by SeaOfOcean — APPROVED** [review]

**Review by charles9304 — APPROVED** [review]

**Comment by Fanoid** [comment]
LGTM. Just a small question: since V4 is very similar with V1V2, is it possible and necessary to merge them together and only maintain their differences?

**Pull request merged**